frontend/packages/console-shared/src/hooks/useBuildConfigsWatcher.ts

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { CronJobKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { getBuildConfigsForCronJob, getBuildConfigsForResource } from '../utils';
+import { BuildConfigOverviewItem } from '../types';
+
+export const useBuildConfigsWatcher = (
+  resource: K8sResourceKind,
+): { loaded: boolean; loadError: string; buildConfigs: BuildConfigOverviewItem[] } => {
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
+  const [buildConfigs, setBuildConfigs] = React.useState<BuildConfigOverviewItem[]>();
+  const { namespace } = resource.metadata;
+  const watchedResources = React.useMemo(
+    () => ({
+      buildConfigs: {
+        isList: true,
+        kind: 'BuildConfig',
+        namespace,
+      },
+      builds: {
+        isList: true,
+        kind: 'Build',
+        namespace,
+      },
+    }),
+    [namespace],
+  );
+
+  const resources = useK8sWatchResources(watchedResources);
+
+  React.useEffect(() => {
+    const errorKey = Object.keys(resources).find((key) => resources[key].loadError);
+    if (errorKey) {
+      setLoadError(resources[errorKey].loadError);
+      return;
+    }
```

Comment:
Small detail, but can simplified with Object.values:

```suggestion
    const resourceWithLoadError = Object.values(resources).find((r) => r.loadError);
    if (resourceWithLoadError) {
      setLoadError(resourceWithLoadError.loadError);
      return;
    }
```

For me both is fine, doesn't block this PR for me.